SpatialGDK/Source/SpatialGDKFunctionalTests/Private/SpatialFunctionalTestRequireHandler.cpp

```diff
@@ -81,267 +81,366 @@ SpatialFunctionalTestRequireHandler::SpatialFunctionalTestRequireHandler()
 	}
+	else
+	{
+		SuccessMsg = FString::Printf(TEXT("Received value %s, not equal to %s, as expected"), *Value, *NotExpected);
+	}
 
-	return GenericRequire(Msg, bPassed, ErrorMsg);
+	return GenericRequire(Msg, bPassed, ErrorMsg, SuccessMsg);
 }
 
 bool SpatialFunctionalTestRequireHandler::RequireNotEqual(const FName& Value, const FName& NotExpected, const FString& Msg)
 {
 	bool bPassed = Value != NotExpected;
 	FString ErrorMsg;
+	FString SuccessMsg;
 
 	if (!bPassed)
 	{
 		ErrorMsg = FString::Printf(TEXT("Received %s but wasn't expecting it"), *Value.ToString());
 	}
+	else
+	{
+		SuccessMsg = FString::Printf(TEXT("Received value %s, not equal to %s, as expected"), *Value.ToString(), *NotExpected.ToString());
+	}
 
-	return GenericRequire(Msg, bPassed, ErrorMsg);
+	return GenericRequire(Msg, bPassed, ErrorMsg, SuccessMsg);
 }
 
 bool SpatialFunctionalTestRequireHandler::RequireNotEqual(const FVector& Value, const FVector& NotExpected, const FString& Msg)
 {
 	bool bPassed = Value != NotExpected;
 	FString ErrorMsg;
+	FString SuccessMsg;
 
 	if (!bPassed)
 	{
 		ErrorMsg = FString::Printf(TEXT("Received (%s) but wasn't expecting it"), *Value.ToString());
 	}
+	else
+	{
+		SuccessMsg = FString::Printf(TEXT("Received value %s, not equal to %s, as expected"), *Value.ToString(), *NotExpected.ToString());
+	}
 
-	return GenericRequire(Msg, bPassed, ErrorMsg);
+	return GenericRequire(Msg, bPassed, ErrorMsg, SuccessMsg);
 }
 
 bool SpatialFunctionalTestRequireHandler::RequireNotEqual(const FRotator& Value, const FRotator& NotExpected, const FString& Msg)
 {
 	bool bPassed = Value != NotExpected;
 	FString ErrorMsg;
+	FString SuccessMsg;
 
 	if (!bPassed)
 	{
 		ErrorMsg = FString::Printf(TEXT("Received (%s) but wasn't expecting it"), *Value.ToString());
 	}
+	else
+	{
+		SuccessMsg = FString::Printf(TEXT("Received value %s, not equal to %s, as expected"), *Value.ToString(), *NotExpected.ToString());
+	}
 
-	return GenericRequire(Msg, bPassed, ErrorMsg);
+	return GenericRequire(Msg, bPassed, ErrorMsg, SuccessMsg);
 }
 
 bool SpatialFunctionalTestRequireHandler::RequireNotEqual(const FTransform& Value, const FTransform& NotExpected, const FString& Msg)
 {
 	bool bPassed = !Value.Equals(NotExpected);
 	FString ErrorMsg;
+	FString SuccessMsg;
 
 	if (!bPassed)
 	{
 		ErrorMsg = FString::Printf(TEXT("Received {%s} but wasn't expecting it"), *GetTransformAsString(Value));
 	}
+	else
+	{
+		SuccessMsg = FString::Printf(TEXT("Received value %s, not equal to %s, as expected"), *GetTransformAsString(Value),
+									 *GetTransformAsString(NotExpected));
+	}
 
-	return GenericRequire(Msg, bPassed, ErrorMsg);
+	return GenericRequire(Msg, bPassed, ErrorMsg, SuccessMsg);
 }
 
-bool SpatialFunctionalTestRequireHandler::GenericRequire(const FString& Msg, bool bPassed, const FString& ErrorMsg)
+bool SpatialFunctionalTestRequireHandler::GenericRequire(const FString& Msg, bool bPassed, const FString& ErrorMsg,
```

Comment:
Do we need two message strings? Why not have one message string that we fill in with the error / success message as appropriate?

Reply:
I like this, thoughts from anyone else?

Reply:
Pushed a commit to address this